pkg/metadata/process.go

```diff
@@ -25,28 +27,68 @@ func Process() Provider {
 	return &StatelessProvider{"process", func(pid int) (model.LabelSet, error) {
 		p, err := procfs.NewProc(pid)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("failed to instantiate procfs for PID %d: %w", pid, err)
 		}
 
+		cgroups, err := p.Cgroups()
+		if err != nil {
+			return nil, fmt.Errorf("failed to get cgroups for PID %d: %w", pid, err)
+		}
+
+		cgroup := findFirstCPUCgroup(cgroups)
+
 		comm, err := p.Comm()
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("failed to get comm for PID %d: %w", pid, err)
 		}
 
 		executable, err := p.Executable()
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("failed to get executable for PID %d: %w", pid, err)
 		}
 
 		stat, err := p.Stat()
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("failed to get stat for PID %d: %w", pid, err)
 		}
 
 		return model.LabelSet{
-			"comm":       model.LabelValue(comm),
-			"executable": model.LabelValue(executable),
-			"ppid":       model.LabelValue(strconv.Itoa(stat.PPID)),
+			"cgroup_name": model.LabelValue(cgroup.Path),
+			"comm":        model.LabelValue(comm),
+			"executable":  model.LabelValue(executable),
+			"ppid":        model.LabelValue(strconv.Itoa(stat.PPID)),
 		}, nil
 	}}
 }
+
+func findFirstCPUCgroup(cgroups []procfs.Cgroup) procfs.Cgroup {
+	// If only 1 cgroup, simply return it
+	if len(cgroups) == 1 {
+		return cgroups[0]
+	}
+
+	for _, cg := range cgroups {
+		// Find first cgroup v1 with cpu controller
+		for _, ctlr := range cg.Controllers {
+			if ctlr == "cpu" {
+				return cg
+			}
+		}
+
+		// Find first systemd slice
+		// https://systemd.io/CGROUP_DELEGATION/#systemds-unit-types
+		if strings.HasPrefix(cg.Path, "/system.slice/") || strings.HasPrefix(cg.Path, "/user.slice/") {
+			return cg
+		}
+
+		// FIXME: what are we looking for here?
+		// https://systemd.io/CGROUP_DELEGATION/#controller-support
+		for _, ctlr := range cg.Controllers {
+			if strings.Contains(ctlr, "systemd") {
```

Comment:
Yes, this conflicts with the documentation. Is there a version difference? Maybe `unified` mode.
Apparently this is an edge case since this is the second time we check the controllers.

Reply:
Unified mode is cgroup v2, this is always empty in v2 

> For cgroups version 1 hierarchies, this field contains a comma-separated list of the controllers bound to the hierarchy.  For the cgroups version 2 hierarchy, this field is empty.

https://man7.org/linux/man-pages/man7/cgroups.7.html